giskard_vision/core/scanner/scanner.py

```diff
@@ -129,6 +165,11 @@ def get_detectors(self, tags: Optional[Sequence[str]] = None) -> Sequence:
         detectors = []
         classes = DetectorRegistry.get_detector_classes(tags=tags)
 
+        # Filter vision detector classes
+        vision_classes = DetectorRegistry.get_detector_classes(tags=["vision"])
+        keys_to_keep = set(vision_classes.keys()).intersection(classes.keys())
+        classes = {k: classes[k] for k in keys_to_keep}
+
         # Filter detector classes
         if self.only:
             only_classes = DetectorRegistry.get_detector_classes(tags=self.only)
```

Comment:
I would merge these two. 

right now if the self.only contains a tag that is not "vision" it will be picked up, as you're running get_detector_classes again. 

I would do something like:
```py
vision_classes = DetectorRegistry.get_detector_classes(tags=["vision"] + self.only if self.only else [])
```